### PR TITLE
Add prefix support for AWS secrets fetch

### PR DESCRIPTION
**Pull request by usmansaleem**
Part of #499.

### CODE REVIEW

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java [hunk 1]
```diff
@@ -138,16 +141,70 @@ void emptyTagFiltersReturnAllSecrets() {
...
+            SECRET_NAME1_KEY1_VALA,
+            SECRET_NAME2_KEY1_VALB,
+            SECRET_NAME3_KEY2_VALC,
+            SECRET_NAME4_KEY2_VALB)
+        .allMatch(name -> name.startsWith(SECRET_NAME_PREFIX));
+    assertThat(secretEntries.stream().map(SimpleEntry::getValue))
+        .contains(SECRET_VALUE1, SECRET_VALUE2, SECRET_VALUE3, SECRET_VALUE4);
+  }
+
   @Test
```
Comment (jframe) [inline keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:203]: Can we add test for when a prefix is combined with a tag

Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:212]: done

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java [hunk 2]
```diff
@@ -138,16 +150,84 @@ void emptyTagFiltersReturnAllSecrets() {
...
+            SECRET_NAME4_KEY2_VALB);
+    assertThat(secretEntries.stream().map(SimpleEntry::getValue))
+        .contains(SECRET_VALUE1, SECRET_VALUE2, SECRET_VALUE3, SECRET_VALUE4);
+  }
+
+  @Test
+  void nonExistentPrefixFilterReturnsEmpty() {
+    final Collection<SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            List.of("nonexistent-secret-prefix"),
```
Comment (siladu) [inline keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:184]: If we combine nonexistent-secret-prefix with a tag that exists, do we still get empty? i.e. are the filters combined with an 'AND'?

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java [hunk 3]
```diff
@@ -138,16 +150,84 @@ void emptyTagFiltersReturnAllSecrets() {
...
+
+    assertThat(secretEntries.stream().map(SimpleEntry::getKey))
+        .contains(
+            SECRET_NAME1_KEY1_VALA,
+            SECRET_NAME2_KEY1_VALB,
+            SECRET_NAME3_KEY2_VALC,
+            SECRET_NAME4_KEY2_VALB)
+        .allMatch(name -> name.startsWith(SECRET_NAME_PREFIX));
+    assertThat(secretEntries.stream().map(SimpleEntry::getValue))
+        .contains(SECRET_VALUE1, SECRET_VALUE2, SECRET_VALUE3, SECRET_VALUE4);
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:209]: will attempt that 👍🏼.

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java (hunk 2 above)
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:184]: It is an AND. I tested it but didn't add a test, I'll add it now.

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/SecretValue.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.aws;
+
+import java.util.Objects;
+
+public class SecretValue {
+  private String secretValue;
+  private String tagKey;
+  private String tagValue;
```
Comment (jframe) [inline keystorage/aws/src/test/java/tech/pegasys/signers/aws/SecretValue.java:20]: nit: these can be final

**Review by jframe — APPROVED** [review]
LGTM

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/SecretValue.java (hunk 4 above)
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/SecretValue.java]: good catch, done!

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java (hunk 2 above)
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:190]: done

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java [hunk 5]
```diff
@@ -138,16 +150,84 @@ void emptyTagFiltersReturnAllSecrets() {
...
+  void listAndMapSecretsWithPrefixAndTags() {
+    final Collection<SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            List.of(SECRET_NAME_PREFIX),
+            List.of("tagKey1"),
+            Collections.emptyList(),
+            SimpleEntry::new);
+
+    assertThat(secretEntries.stream().map(SimpleEntry::getKey))
+        .contains(SECRET_NAME1_KEY1_VALA, SECRET_NAME2_KEY1_VALB)
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java]: updated test to check that non expected values are not returned.

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java]: Updated test case to validate that non-prefixed secrets are not returned.

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java [hunk 6]
```diff
@@ -97,128 +114,269 @@ void teardown() {
...
+        secretsMaps.getPrefixASecretsMap().entrySet().stream()
+            .filter(v -> Objects.equals("tagKey1", v.getValue().getTagKey()))
+            .map(Map.Entry::getKey)
+            .collect(Collectors.toSet());
+
+    final Set<String> nonExpectedKeys =
+        secretsMaps.getPrefixASecretsMap().entrySet().stream()
+            .filter(v -> !Objects.equals("tagKey1", v.getValue().getTagKey()))
+            .map(Map.Entry::getKey)
+            .collect(Collectors.toSet());
```
Comment (siladu) [inline keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java:237]: nit/suggestion: could use `Collectors.partitioningBy()`

**Review by siladu — APPROVED** [review]
LGTM, just nits :)

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java (hunk 6 above)
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java]: good hint, updated!

File: keystorage/aws/src/test/java/tech/pegasys/signers/aws/SecretsMaps.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  public static String computeMapBKey(final int i) {
+    return String.format("%ssecret%d", SECRET_NAME_PREFIX_B, i);
+  }
+
+  public static SecretValue computeSecretValue(final int i) {
+    final String value = String.format("secret-value%d", i);
+    final SecretValue secretValue;
+    // due to past test values setup ...
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply keystorage/aws/src/test/java/tech/pegasys/signers/aws/SecretsMaps.java]: removed the comment as it was adding confusion.

**Pull request merged**